### PR TITLE
Fix for blacklist ids bug and added pagination

### PR DESCRIPTION
**Pull request by powahftw**

### CODE REVIEW

File: static/css/style.css [hunk 1]
```diff
@@ -318,6 +318,17 @@ ul {
   margin: 0;
 }
 
+.pagination-container button {
```
Comment (mvkv) [inline static/css/style.css:321]: Add cursor pointer and change the colors of the arrows to white

File: flask_server.py [hunk 2]
```diff
@@ -208,9 +209,13 @@ def logout():
...
 
-@app.route('/api/logs/', methods=["GET"])
-def get_logs():
-    return jsonify({"logs": get_system_logs()})
+@app.route("/api/logs/<int:page>/", methods=['GET'])
+def get_logs(page):
+    logs = get_app_logs()
+    paginated_logs = [logs[start:start+PAGINATE_EVERY_N_ROWS] for start in range(0, len(logs), PAGINATE_EVERY_N_ROWS)]
+    if page < 1 or page > len(paginated_logs):
+        return jsonify({"page": -1, "max_page": len(paginated_logs), "logs": []})
```
Comment (mvkv) [inline flask_server.py:217]: I think these two lines are useless

File: static/js/logs.js [hunk 3]
```diff
@@ -12,9 +13,34 @@ const renderLogs = (logLines) => {
...
+    root.append(decrease);
+  }
+  root.append(document.createTextNode(`${pageN} / ${maxPage}`));
+  if (pageN < maxPage) {
+    increase = document.createElement('button');
+    increase.innerText = '>';
+    increase.addEventListener('click', () => getAndRenderLogs(pageN + 1));
+    root.append(increase);
+  }
+  paginationNode.append(root);
```
Comment (mvkv) [inline static/js/logs.js]: nit: fix the spacing when there's only one arrow

**Review by mvkv — APPROVED** [review]

File: static/css/style.css (hunk 1 above)
Comment (mvkv) [reply static/css/style.css:321]: Done

File: flask_server.py (hunk 2 above)
Comment (mvkv) [reply flask_server.py:217]: Resolved client side

File: static/js/logs.js (hunk 3 above)
Comment (mvkv) [reply static/js/logs.js]: Done

**Review by mvkv — APPROVED** [review]

File: static/js/settings.js [hunk 4]
```diff
@@ -189,6 +189,11 @@ const setUpButtonsListeners = () => {
   const submitBtn = document.getElementById('submitBtn');
   const addExtraIDBtn = document.getElementById('add-extra-id');
   const addBlacklistedIDBtn = document.getElementById('add-blacklisted-id');
+  const changingInputs = [
```
Comment (powahftw) [inline static/js/settings.js]: Move this below, right above where you use them, so it's logically closer. You could also just store the id and add the document.getElementById in the loop.

File: static/js/logs.js [hunk 5]
```diff
@@ -15,9 +15,9 @@ const renderLogs = (logLines) => {
...
 const renderNav = (pageN, maxPage) => {
   const paginationNode = document.getElementById('pagination');
-  paginationNode.innerHTML = ''
+  paginationNode.innerHTML = '';
 
-  if (maxPage == 1 || maxPage == 0) {
+  if (maxPage <= 1) {
     return; // Don't display pagination on log files that fit in a single page.
   }
   const root = document.createElement('div');
```
Comment (powahftw) [inline static/js/logs.js:23]: Move the buttons in the HTML if they are always rendered. Add an id (previous-page, next-page) to attach the event listener. Make them invisible by default.

**Pull request merged**